runtime/api/Robot.py

```diff
@@ -12,49 +12,494 @@
+    the result from one specific color sensor.
+
+    :param name: A string that identifies the color sensor
+    :param color: A integer that identifies which specific color sensor to return
+                  where 0 specifies the red sensor, 1 specifies the green sensor, 
+                  and 2 specifies the blue sensor
+    :returns: A double between 0 and 1, where 1 indicates a higher intensity
+
+    :Examples:
+
+    >>> color = get_color_sensor("color1",1)
+    >>> color
+    0.873748
+
+    """
+    all_data = mc.get('sensor_values')
+    name = _lookup(name)
+    try:
+        return all_data[name][color]
+    except KeyError:
+        raise KeyError("No sensor with that name")
+
+def get_luminosity(name):
+    """Returns the luminosity for the specified color sensor.
+
+    The color sensor returns the luminosity detected by the color sensor, represnted by
+    a decimal between 0 and 1, where 1 indicates higher intensity.
+
+    :param name: A string that identifies the color sensor
+    :returns: A double between 0 and 1, where 1 indicates a higher intensity
+
+    :Examples:
+
+    >>> get_luminosity("color1")
+    0.89783
+
+    """
+    all_data = mc.get('sensor_values')
+    name = _lookup(name)
+    try:
+        return all_data[name][3]
+    except KeyError:
+        raise KeyError("No sensor with that name")
+
+def get_hue(name):
+    """Returns the hue detected at the specified color sensor.
+
+    This uses the red, green, and blue sensors on the color sensor to return the 
+    detected hue, which is represented as a double between 0 and 360. See 
+    https://en.wikipedia.org/wiki/Hue for more information on hue.
+
+    :param name: A string that identifies the color sensor
+    :returns: A double between 0 and 360 representing the detected hue
+
+    :Examples:
+
+    >>> hue = get_hue("color1")
+    >>> hue
+    254.01
+
+    """
+    all_data = mc.get('sensor_values')
+    name = _lookup(name)
+    try:
+        r = all_data[name][0]
+        g = all_data[name][1]
+        b = all_data[name][2]
+        denom = max(r,g,b) - min(r,g,b)
+        if r > g and r > b:
+            return (g - b)/denom
+        elif g > r and g > b:
+            return 2.0 + (b - r)/denom
+        else:
+            return 4.0 + (r - g)/denom
+    except KeyError:
+        raise KeyError("No Sensor with that name")
+
+def get_distance_sensor(name):
+    """Returns the distance away from the sensor an object is, measured in centimeters
+
+    :param name: A String that identifies the distance sensor
+    :returns: A double representing how many centimeters away the object is from the sensor
+
+    :Examples: 
+
+    >>> distance = get_distance_sensor("distance1")
+
+    """
+    all_data = mc.get('sensor_values')
+    name = _lookup(name)
+    try:
+        return all_data[name][0]
+    except KeyError:
+        raise KeyError("No Sensor with that name")
+
+def get_all_switches(name):
```

Comment:
Our thinking behind this was that each type of sensor returned a different form of value that meant different things (e.g. limit switches returned 4 different values since each smart device had four attached). We realize that the code is virtually the same for each method, but this was a way to explain/tell the student what to expect from each sensor.